articles/native-platforms/ionic/04-linking-accounts.md

```diff
@@ -0,0 +1,257 @@
+
+To differentiate the login from the linking login, you will create a second instance of `AuthLock` to obtain the secondary account JWT.
+
+```js
+/* ===== www/components/auth/auth.service.js ===== */
+(function () {
+
+	...
+  
+  function authService($rootScope, lock, authManager, jwtHelper, $http, $q) {
+
+	...
+
+    function linkAccount() {
+      try {
+        var profile = JSON.parse(localStorage.getItem('profile'));
+        var token = localStorage.getItem('id_token');
+      } catch (e) {
+        return false;
+      }
+
+      var options = {
+        rememberLastLogin: false,
+        auth: {
+          redirect: false,
+          params: {
+            scope: 'openid',
+            device: 'Mobile device'
+          }
+        }
+      };
+
+      var lockLink = new Auth0Lock(AUTH0_CLIENT_ID, AUTH0_DOMAIN, options);
+      var deferred = $q.defer();
+
+      lockLink.on('authenticated', function (authResult) {
+
+		// do linking accounts
+
+      });
+
+      lockLink.show();
+
+      return deferred.promise;
+
+    }
+
+    return {
+      ...
+	  
+      linkAccount: linkAccount,
+	  
+      ...
+    }
+  }
+})();
+```
+
+Now that the second login is handled, you will need to actually do the linking.
+
+```js
+/* ===== www/components/auth/auth.service.js ===== */
+
+
+lockLink.on('authenticated', function (authResult) {
+ 
+    $http({
+      method: 'POST',
+      url: 'https://' + AUTH0_DOMAIN + '/api/v2/users/' + profile.user_id + '/identities',
+      headers: {
+        Authorization: 'Bearer ' + token
+      },
+      data: {
+        link_with: authResult.idToken
+      }
+    })
+      .then(function () {
+        lockLink.hide();
+   
+        lock.getProfile(token, function (error, profile) {
+          if (!error) {
+            deferred.resolve(profile);
+          } else {
+            deferred.reject(error);
+          }
+        });
+   
+      });
+
+});  
+```
+
+The function posts to the API, passing the `link_with` parameter with the JWT value in the body. Then it fetches the profile on success to check that the accounts are now linked.
+
+Now to begin the link process, call `linkAccount` method and subscribe on it to update user profile:
+
+```js
+/* ===== www/components/home/home.controller.js ===== */
+(function () {
+
+	...
+  function HomeController($state, authService, $scope) {
+    ...
+
+    function linkAccount() {
+      authService.linkAccount()
+        .then(function (profile) {
+          vm.profile = profile;
+          localStorage.setItem('profile', JSON.stringify(profile));
+          refreshIdentities();
+        })
+    }
+
+    ...
+
+  }
+
+}());
+
+```
+
+## User Profile Linked Accounts Information
+
+The user profile contains an array of identities which includes the profile information from linked providers. 
+
+To view a user's identities, access the [Users](${uiURL}/#/users) page on the Auth0 dashboard, select a user, and scroll down to `identities`. 
```

Comment:
@lobo `uiURL` is not used any longer, so i throws `ReferenceError: uiURL is not defined`. Replace it with `manage_url`.